public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/AddNotificationChannelModal.test.tsx

```diff
@@ -1,15 +1,59 @@
 import React from 'react';
-import { mount } from 'enzyme';
+import { mount, ReactWrapper } from 'enzyme';
 import { act } from 'react-dom/test-utils';
 import { dataQa } from '@percona/platform-core';
 import { AddNotificationChannelModal } from './AddNotificationChannelModal';
 import { TYPE_OPTIONS } from './AddNotificationChannel.constants';
+import { NotificationChannelType } from '../NotificationChannel.types';
 import { notificationChannelStubs } from '../__mocks__/notificationChannelStubs';
 
 jest.mock('../NotificationChannel.service');
 jest.mock('app/core/app_events');
 
-describe('AddNotificationChannelModal', () => {
+/**
+ * Return this form's "Add" button
+ */
+const findFormButton = (wrapper: ReactWrapper): ReactWrapper =>
+  wrapper
+    .find(dataQa('notification-channel-add-button'))
+    .find('button')
+    .at(0);
+
+/**
+ * Given a type, return the matching element.
+ * The element returned is the one that's actually clickable and that triggers the change
+ */
+const findTypeOption = (wrapper: ReactWrapper, type: NotificationChannelType): ReactWrapper => {
+  const menuOptions = wrapper.find({ 'aria-label': 'Select option' });
+  const matchingOption = TYPE_OPTIONS.find(option => option.value === type);
+
+  return menuOptions.filterWhere(node => node.text() === matchingOption.label);
+};
+
+/**
+ * Fill the form with a name and a notification channel type
+ */
+const fillNameAndType = (wrapper: ReactWrapper, type: NotificationChannelType, name = 'John Doe') => {
+  const inputs = wrapper.find('input');
+  const nameInput = inputs.at(0);
+  const typeInput = inputs.at(1);
```

Comment:
nit: dataQa can be used here as well (you use it in this test already)

Reply:
There's a catch here: we're using the  Grafana's `Select` component, which does not allow us to pass the `data-qa` all the way down to the input. Should I only modify the `nameInput` to use `dataQa(...)`? Wouldn't it be better to just take advantage of the fact that the inputs are already there, instead of making another lookup?

Reply:
Ok, we need a better (our own Select then))

Reply:
> There's a catch here: we're using the Grafana's `Select` component, which does not allow us to pass the `data-qa` all the way down to the input. Should I only modify the `nameInput` to use `dataQa(...)`? Wouldn't it be better to just take advantage of the fact that the inputs are already there, instead of making another lookup?

In terms of execution speed, there is no difference here. I'm all for consistency.

Reply:
Yep, we need our own indeed :))